exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceStreamObservers.java

```diff
@@ -0,0 +1,245 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.ocagent;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.stub.StreamObserver;
+import io.opencensus.proto.agent.trace.v1.CurrentLibraryConfig;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceRequest;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceResponse;
+import io.opencensus.proto.agent.trace.v1.UpdatedLibraryConfig;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.config.TraceParams;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+*/
+
+/** Stream observers of OC-Agent Trace Service. */
+final class OcAgentTraceServiceStreamObservers {
+
+  private OcAgentTraceServiceStreamObservers() {}
+
+  private static final Logger logger =
+      Logger.getLogger(OcAgentTraceServiceStreamObservers.class.getName());
+
+  /*
+   * We should use ManagedChannel.getState() to determine the state of current connection.
+   *
+   * // TODO(songya): avoid going into too much details of each stream and use generic
+   * request/response if that's more clear.
+   *
+   * 0. For export RPC we hold a reference "exportRequestObserverRef" to a StreamObserver for
+   * request stream, and "exportResponseObserverRef" to a StreamObserver for response stream.
+   * For config RPC we have "currentConfigObserverRef" and "updatedConfigObserverRef". Note that
+   * exportRequestObserver and currentConfigObserver are defined by Agent, while
+   * exportResponseObserver and updatedConfigObserver are defined in this class.
+   *
+   * Think of the whole process as a Finite State Machine.
+   *
+   * Note: "Exporter" = Agent Exporter and "Agent" = OC-Agent in the sections below.
+   *
+   * STATE 1: Unconnected/Disconnected.
+   *
+   * 1. First, Exporter will try to establish the streams by calling TraceServiceStub.export and
+   * TraceServiceStub.config in a daemon thread.
+   *
+   *   1-1. If the initial attempt succeeded, references of exportRequestObserver and
+   *   currentConfigObserver should be returned to Exporter as callbacks. We keep a reference of
+   *   them in exportRequestObserverRef and currentConfigObserverRef. At the same time, references
+   *   of exportRequestObserver and updatedConfigObserver will be passed to Agent as callbacks.
```

Comment:
As I mentioned before you don't pass any callback. The stream object is owned by the Exporter, you can say via these streams the Exporter receives messages from the Agent. Similar for request streams they are used to send messages to the Agent.

Reply:
Rephrased a bit.